shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/API/Validation.hs

```diff
@@ -31,18 +34,147 @@ import GHC.Generics (Generic)
 import NoThunks.Class (NoThunks (..))
 import Shelley.Spec.Ledger.BaseTypes (Globals (..))
 import Shelley.Spec.Ledger.BlockChain
+import Shelley.Spec.Ledger.Keys (DSignable)
 import qualified Shelley.Spec.Ledger.LedgerState as LedgerState
 import qualified Shelley.Spec.Ledger.STS.Bbody as STS
 import qualified Shelley.Spec.Ledger.STS.Chain as STS
 import qualified Shelley.Spec.Ledger.STS.Tick as STS
 import Shelley.Spec.Ledger.Slot (SlotNo)
+import Shelley.Spec.Ledger.TxBody (EraIndependentTxBody)
+
+{-------------------------------------------------------------------------------
+  Block validation API
+-------------------------------------------------------------------------------}
 
 -- | Type alias for the state updated by TICK and BBODY rules
 type ShelleyState = LedgerState.NewEpochState
 
+class
+  ( Eq (Block era),
+    Show (Block era),
+    NoThunks (Block era),
+    FromCBOR (Annotator (Block era)),
+    ToCBOR (Block era),
+    Eq (BHeader (Crypto era)),
+    Show (BHeader (Crypto era)),
+    NoThunks (BHeader (Crypto era)),
+    FromCBOR (Annotator (BHeader (Crypto era))),
+    ToCBOR (Block era),
+    Eq (ShelleyState era),
+    Show (ShelleyState era),
+    NoThunks (ShelleyState era),
+    FromCBOR (ShelleyState era),
+    ToCBOR (ShelleyState era),
+    Eq (BlockTransitionError era),
+    Show (BlockTransitionError era),
+    NoThunks (BlockTransitionError era),
+    Eq (STS.PredicateFailure (STS.CHAIN era)),
+    Show (STS.PredicateFailure (STS.CHAIN era)),
+    NoThunks (STS.PredicateFailure (STS.CHAIN era))
+  ) =>
+  ApplyBlock era
+  where
+  -- | Apply the header level ledger transition.
+  --
+  -- This handles checks and updates that happen on a slot tick, as well as a
+  -- few header level checks, such as size constraints.
+  applyTick ::
+    Globals ->
+    ShelleyState era ->
```

Comment:
[bikeshedding] `ShelleyState` was an alias for downstream use, but now feels a bit awkward, since generally this won't be shelley. We could use the wrapped type `NewEpochState`, or come up with a better name (but all the obvious ones are taken...)

Reply:
I have gone for `NewEpochState` for now.